src/core/aws-server-http_utils.adb

```diff
@@ -1716,7 +1732,11 @@ package body AWS.Server.HTTP_Utils is
                         Send_WebSocket_Handshake_Error
                           (Messages.S403,
                            Exception_Message (E));
+
+                        --  ??? Is the socket automatically unregistered
```

Comment:
No it is not automatically unregistered, you need to do it.

Reply:
I changed the code to unregister when needed. The behavior of `Registry.Close` is a bit surprising because it receives an `Object` as parameter, but will internally lookup the pointer and free it. So the pointer I am passing (`WS`) is freed underneath although I never passed it. But I think this is related to #209 so let's not worry about that here